stubs/tensorflow/tensorflow/keras/metrics.pyi

```diff
@@ -1,9 +1,136 @@
-from tensorflow import Tensor
-from tensorflow._aliases import TensorCompatible
+from _typeshed import Incomplete
+from abc import ABCMeta, abstractmethod
+from collections.abc import Callable, Iterable, Sequence
+from typing import Any, Literal
+from typing_extensions import Self, TypeAlias, override
 
+import tensorflow as tf
+from tensorflow import Operation, Tensor
+from tensorflow._aliases import DTypeLike, KerasSerializable, TensorCompatible
+from tensorflow.keras.initializers import _Initializer
+from tensorflow.keras.layers import _Constraint
+from tensorflow.keras.regularizers import _Regularizer
+
+_Output: TypeAlias = Tensor | dict[str, Tensor]
+
+class Metric(tf.keras.layers.Layer[tf.Tensor, tf.Tensor], metaclass=ABCMeta):
+    def __init__(self, name: str | None = None, dtype: DTypeLike | None = None) -> None: ...
+    def __new__(cls, *args: Any, **kwargs: Any) -> Self: ...
+    def __call__(self, *args: Incomplete, **kwargs: Incomplete) -> Incomplete: ...
+    def __deepcopy__(self, memo: Incomplete | None = None) -> Incomplete: ...
+    @property
+    def _trackable_saved_model_saver(self) -> Incomplete: ...
+    @property
+    def trainable_weights(self) -> list[Incomplete]: ...
```

Comment:
? I think this method is inherited from Layer. Only methods defined new in class should be covered while inherited ones should be defined in parent.

Reply:
Done in 8fd3fad1c